File: src/test/java/com/raychenon/leetcode/EditDistanceTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 1/6/2020
 * https://leetcode.com/problems/edit-distance
 */
public class EditDistanceTest {

    private static EditDistance editDistance;

    @BeforeClass
    public static void setUp() {
        editDistance = new EditDistance();
    }

    @Test
    public void example1Test() {
        assertEditDistance(3, "horse", "ros");
    }

    @Test
    public void example2Test() {
        assertEditDistance(5, "intention", "execution");
    }

    @Test
    public void example3Test() {
        assertEditDistance(10, "zoologicoarchaeologist", "zoogeologist");
    }

    void assertEditDistance(int expected, String word1, String word2) {
        assertEquals(expected, editDistance.minDistance(word1, word2));
        assertEquals(expected, editDistance.minDistance1DArrayMemory(word1, word2));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/EditDistanceTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 1/6/2020
 * https://leetcode.com/problems/edit-distance/
 */
class EditDistanceTest {

    @Test
    fun example1Test() {
        assertEditDistance(3, "horse", "ros")
    }

    @Test
    fun example2Test() {
        assertEditDistance(5, "intention", "execution")
    }

    @Test
    fun example3Test() {
        assertEditDistance(10, "zoologicoarchaeologist", "zoogeologist")
    }

    fun assertEditDistance(expected: Int, word1: String, word2: String) {
        TestCase.assertEquals(expected, EditDistance.minDistance(word1, word2))
    }
}